### PR TITLE
Feature/andrewzpu/program page

### LINKED ISSUE #31: [View All Programs Page] - Create Program card components to show all necessary information about a program
## Description

The programs are organized in a unique way in that they are in cards kind of like canvas. We need these card components in order to be able to populate the program page

## Requirements

- [ ] Card contains all information as demonstrated on figma
- [ ] Edit program ellipses should be disabled when team member is viewing it (just make it an optional field)

### LINKED ISSUE #44: [View All Programs Page] - Create main UI page that has all programs populated with add new program button on right hand side

### LINKED ISSUE #47: [View All Programs Page] - Create a backend route to get all programs

### PR DESCRIPTION
**Pull request by andrewzpu**
## Tracking Info

Resolves #31
Resolves #44 
Resolves #47 

## Changes

Created layout Programs Card UI element and Programs page.
Implemented different layouts for different screen sizes
Added some more settings to tailwind config and global.css

Created backend route to get list of all existing programs.

Connected programs page to backend route to visually display all program data

## Testing

Ran on my own local host
Tested different size screens using device toolbar in developer view

## Confirmation of Change

Standard View
image

Tablet View
image

Mobile View
image

Extra Large View
image

Backend Route Result
image

Programs page with database program data populated
image

Notes:
Card color in populated data is saved as #AEFFEA, this is not the default color
Student counts for each card is currently commented out until this data is added to the program schema

### CODE REVIEW

**Review by mraysu — APPROVED** [review]
Looks clean!

File: frontend/src/api/requests.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * Custom type definition for the HTTP methods handled by this module.
  */
 type Method = "GET" | "POST" | "PUT" | "PATCH" | "DELETE";
-const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL ?? "/api";
+const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL ?? "http://localhost:4000/api";
```
Comment (adhi0331) [inline frontend/src/api/requests.ts]: This needs to be what it was previously `const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL ?? "/api";`. In your frontend env file add the following variable `NEXT_PUBLIC_API_BASE_URL='http://localhost:4000/api'`

**Review by adhi0331 — CHANGES_REQUESTED** [review]
One minor change

File: frontend/src/api/requests.ts (hunk 1 above)
Comment (andrewzpu) [reply frontend/src/api/requests.ts]: Ok, I just changed it back

**Review by adhi0331 — APPROVED** [review]
LGTM! 🚀

**Pull request merged**